replica-node/src/main.rs

```diff
@@ -0,0 +1,356 @@
+        .arg(
+            Arg::with_name("entrypoint")
+                .short("n")
+                .long("entrypoint")
+                .value_name("HOST:PORT")
+                .takes_value(true)
+                .multiple(true)
+                .validator(solana_net_utils::is_host_port)
+                .help("Rendezvous with the cluster at this gossip entrypoint"),
+        )
+        .arg(
+            Arg::with_name("bind_address")
+                .long("bind-address")
+                .value_name("HOST")
+                .takes_value(true)
+                .validator(solana_net_utils::is_host)
+                .default_value("0.0.0.0")
+                .help("IP address to bind the replica ports"),
+        )
+        .arg(
+            Arg::with_name("rpc_bind_address")
+                .long("rpc-bind-address")
+                .value_name("HOST")
+                .takes_value(true)
+                .validator(solana_net_utils::is_host)
+                .help("IP address to bind the Json RPC port [default: use --bind-address]"),
+        )
+        .arg(
+            Arg::with_name("rpc_port")
+                .long("rpc-port")
+                .value_name("PORT")
+                .takes_value(true)
+                .validator(solana_validator::port_validator)
+                .help("Enable JSON RPC on this port, and the next port for the RPC websocket"),
+        )
+        .arg(
+            Arg::with_name("dynamic_port_range")
+                .long("dynamic-port-range")
+                .value_name("MIN_PORT-MAX_PORT")
+                .takes_value(true)
+                .default_value(default_dynamic_port_range)
+                .validator(port_range_validator)
+                .help("Range to use for dynamically assigned ports"),
+        )
+        .arg(
+            Arg::with_name("expected_shred_version")
+                .long("expected-shred-version")
+                .value_name("VERSION")
+                .takes_value(true)
+                .validator(is_parsable::<u16>)
+                .help("Require the shred version be this value"),
+        )
+        .arg(
+            Arg::with_name("logfile")
+                .short("o")
+                .long("log")
+                .value_name("FILE")
+                .takes_value(true)
+                .help(
+                    "Redirect logging to the specified file, '-' for standard error. \
+                       Sending the SIGUSR1 signal to the validator process will cause it \
+                       to re-open the log file",
+                ),
+        )
+        .get_matches();
+
+    let bind_address = solana_net_utils::parse_host(matches.value_of("bind_address").unwrap())
+        .expect("invalid bind_address");
+
+    let rpc_bind_address = if matches.is_present("rpc_bind_address") {
+        solana_net_utils::parse_host(matches.value_of("rpc_bind_address").unwrap())
+            .expect("invalid rpc_bind_address")
+    } else {
+        bind_address
+    };
```

Comment:
```suggestion
    let rpc_bind_address = if let Some(rpc_bind_address) = matches.value_of("rpc_bind_address") {
        solana_net_utils::parse_host(rpc_bind_address).expect("invalid rpc_bind_address")
    } else {
        bind_address
    };
```
This is more idiomatic (I know this pattern is copied from `solana-validator`... but should be fixed there too :) )